linkerd/app/inbound/src/lib.rs

```diff
@@ -237,10 +237,7 @@ impl Config {
                         .push(metrics.stack.layer(stack_labels("target"))),
                 ),
             )
-            .instrument(|_: &Target| info_span!("target"))
-            // Prevent the cache's lock from being acquired in poll_ready, ensuring this happens
-            // in the response future. This prevents buffers from holding the cache's lock.
-            .push_oneshot()
```

Comment:
the oneshots here were only necessary due to implementation details of `Cache` (acquiring the lock in `poll_ready`), so they are now removeable.